src/AccessControl/ZopeGuards.py

```diff
@@ -254,8 +254,11 @@ def _check_list_access(name, value):
 # iterator that is known to be safe (as in guarded_enumerate).
 
 
-def guarded_next(iterator):
-    ob = next(iterator)
+def guarded_next(iterator, default=_marker):
+    if default is _marker:
+        ob = next(iterator)
+    else:
+        ob = next(iterator, default)
     if not isinstance(iterator, SafeIter):
         guard(ob, ob)
```

Comment:
`ob is default` is potentially not checked here. I do not think that this is a problem because the caller could have accessed `default` directly, thus, it should already have been properly guarded.

Reply:
Thanks, yes that's also my understanding.